Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/WindowsMixedRealitySpatialMeshObserver.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+#if WMR_ENABLED
+using UnityEngine.XR.WindowsMR;
+#endif // WMR_ENABLED
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    [MixedRealityDataProvider(
+        typeof(IMixedRealitySpatialAwarenessSystem),
+        SupportedPlatforms.WindowsUniversal,
+        "XR SDK Windows Mixed Reality Spatial Mesh Observer",
+        "Profiles/DefaultMixedRealitySpatialAwarenessMeshObserverProfile.asset",
+        "MixedRealityToolkit.SDK")]
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/SpatialAwareness/SpatialAwarenessGettingStarted.html")]
+    public class WindowsMixedRealitySpatialMeshObserver :
+        GenericXRSDKSpatialMeshObserver,
+        IMixedRealityCapabilityCheck
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="registrar">The <see cref="IMixedRealityServiceRegistrar"/> instance that loaded the service.</param>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public WindowsMixedRealitySpatialMeshObserver(
+            IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(spatialAwarenessSystem, name, priority, profile)
+        { }
+
+        /// <inheritdoc/>
+        protected override void ConfigureObserverVolume()
+        {
+            if (SpatialAwarenessSystem == null || XRSDKSubsystemHelpers.MeshSubsystem == null)
+            {
+                return;
+            }
+
+            // Update the observer
+            switch (ObserverVolumeType)
+            {
+                case VolumeType.AxisAlignedCube:
+                    XRSDKSubsystemHelpers.MeshSubsystem.SetBoundingVolume(ObserverOrigin, ObservationExtents);
+                    break;
+#if WMR_ENABLED
+                case VolumeType.Sphere:
+                    // We use the x value of the extents as the sphere radius
```

Comment:
Is this general convention? i.e. I'm curious if someone using this and sets other extends (i.e. x/y/z) and they don't understand why changing y/z don't do anything.

Reply:
This is the convention we used in the existing WMR mesh observer. @davidkline-ms might be able to add more background on the implementation.

Reply:
The original intent was to limit the number of properties in the class (ex: why add an additional float property when we already have 3 of them in the Vector3).  I have not seen feedback re: confusion and it is a documented behavior.

We could, potentially, modify the class to add a sphere radius property that gets/sets the X value of the vector and display custom ux in the inspector.